mlir/include/mlir/Dialect/Linalg/IR/LinalgStructuredOps.td

```diff
@@ -680,6 +680,125 @@ def MatmulOp : LinalgStructuredBase_Op<"matmul", [
     }];
 }
 
+//===----------------------------------------------------------------------===//
+// Contract op.
+//===----------------------------------------------------------------------===//
+
+def ContractOp : LinalgStructuredBase_Op<"contract", [
+               AttrSizedOperandSegments,
+               LinalgContractionOpInterface]> {
+  let summary = [{
+    Perform a contraction on two inputs, accumulating on top of a third.
```

Comment:
"Accumulating _into_ the third" would sound more natural to me.

Below you document that C is used as the initial value.

Reply:
Have adapted this.

Personally, I feel "accumulating _onto_ the third" would be even better at capturing that we are adding _on top of_ the init value and not overwriting it with the result of the contraction. Not a native speaker though, so if "_into_" is preferable, happy to go with that.

Reply:
Also not a native speaker 😄 

AFAIk, all of these forms are grammatically correct. IIRC,  "into" is more Mathematical/academic/formal. That's what I'd recommend for a short description at the top.

Normally I'd try to share a reference, but this tablet is not great for browsing papers on GEMM 😂

Thanks for the update!